python/tk_desktop/desktop_engine_project_implementation.py

```diff
@@ -187,7 +200,7 @@ def start_app(self):
 
         else:  # not engine.has_ui
             # wait for the engine communication channel to shut down
-            self._project_comm.wait_for_shutdown()
+            self._project_comm.join()
```

Comment:
Renamed back to `join` so that compatibility with the previous version is simpler.